src/plugin/managerbase.js

```diff
@@ -312,12 +306,38 @@ define([
                     plugin.createMessage(null, 'The main callback is being called more than once.');
                     plugin.createMessage(null, stackTrace);
                     callback('The main callback is being called more than once!', result);
+                }
+            };
+            try {
+                if (expectsCallback) {
+                    result = await new Promise((resolve, reject) => 
```

Comment:
is Promise supported by all browsers?

Reply:
Good question. How far back do you care to support? https://caniuse.com/#search=promise